docs/0002-atlas-translations-management.rst

```diff
@@ -0,0 +1,144 @@
+
+- Extract the English translation source entries and commit them into the `openedx-translations repository`_
+- The `GitHub Transifex app`_ will automatically commit the translations to the `openedx-translations repository`_
+- Provide a command that utilizes the `atlas`_ tool to pull translations in both development and web server image builds
+
+Microservices, micro-frontends, XBlocks and plugins use as few translation files as possible to makes it easier for Translators
+to locate and translate relevant resources.
+
+`edx-platform`_ translations strings used to be split between 7 files, but this has been deprecated in favor of
+``edx-platform.po`` and ``edx-platform-js.po`` files.
+More details are available in the `edx-platform resources combination decision document`_.
+
+The `openedx-app-ios`_ uses a clean architecture. Therefore, each app has its own i18n strings files. This is
+useful for Developer Experience, but we believe it's not good for Translator Experience because translators
+often are unaware of the application engineering architecture and would like to be able to quickly search for strings
+regardless of their location in the code.
+
+Decision
+********
+
+This decision requires a few changes to the mobile app translation workflow:
+
+- Translation files are no longer comitted directly to the mobie app repositories (`openedx-app-ios`_ / `openedx-app-android`_)
+- Keep the i18n string files split into each clean architecture app without hurting Developer Experience.
+- Identical string IDs across the application will be treated as the same string/translation.
+  For example, ``BUTTON_CLOSE="Close"`` in dashboard is the same ``BUTTON_CLOSE="Close"`` in profile app.
+  Otherwise, a validation script will throw an error and prevent merging the pull request.
+- String files are combined into a single file before uploading it to Transifex
+  (through the `openedx-translations repository`_)
+- Translators can translate the entire app within a single transifex resource
+- Before releasing to App Store, developers need to run a new ``make pull_translations`` command
+- After translations are pulled the translation files are split into each clean architecture application
+  as if it has been individual resource (a new script)
+
+
+String file combination and splitting process
+=============================================
+This is a new process that's being introduced to have the best combination
+of Developer Experience and Translator Experience.
+
+Translators need to have less resources as possible with strings de-duplicated,
+while Engineers need to have clean architecture with strings split into each app to aid maintainability.
+
+Combining the files during ``combine_translations``
+---------------------------------------------------
+
+However, combining the string files to aid Translator Experience in the following manner:
+
+Suppose we have two apps with one string source file in each::
```

Comment:
And here, can we use some other term instead of `apps`? It's confusing a bit.

Reply:
done